Issue 7718 required field not displaying error when edited to become null

dsilvam opened the pull request.

dsilvam committed:
Bubble up the exception so the error shows in UI

* Catch the exception thrown by setContentletProperly and rethrows it
so it bubbles up to the UI

* Add new catch for the specific exception (DotStateException) to the outer try and leaves
the old generic catch (Exception) to mantain old behaviour

issue:#7718

dsilvam committed:
Remove specific exception from method signature

Since 'Exception' is already thrown in the method signature,
there is no need to throw the more specific DotValidationException

issue #7718

brentgriffin commented:
So, why are we throwing DotContentletStateExceptions up the stack but for other Exceptions just returning false?  Not saying it is wrong but wondering what the rationale is.

dsilvam commented:
It is explained in commit message. To maintain old behaviour. We don't want exceptions different than DotStateException to be bubbled up, since they never were. We don't want to stop processing for exceptions that didn't make the process to stop before.

dsilvam committed:
Remove unnecessary error logging and not used return value

Remove try catch because it was only for including additional not needed
logging

issue #7718
